manta-crypto/src/merkle_tree/test/batch_insertion.rs

```diff
@@ -114,3 +114,37 @@ fn test_batch_insertion_forest() {
         |forest, _, leaves| forest.batch_insert(leaves),
     )
 }
+
+/// Tests that [`merge_fork_partial`] returns the same root as [`merge_fork`].
+///
+/// [`merge_fork`]: ForkedTree::merge_fork
+/// [`merge_fork_partial`]: ForkedTree::merge_fork_partial
+#[test]
+fn branch_and_merge_test() {
+    let mut rng = OsRng;
+    let parameters = Parameters::sample(Default::default(), &mut rng);
+    let mut tree = ForkedTree::new(Partial::new(&parameters), &parameters);
+    let number_of_insertions = rng.gen_range(1..(1 << (HEIGHT - 1)) / 2);
+    let insertions = (0..number_of_insertions)
+        .map(|_| rng.gen())
+        .collect::<Vec<_>>();
+    for leaf in insertions {
+        Tree::push(&mut tree, &parameters, &leaf);
+    }
+    tree.merge_fork(&parameters);
+    let second_number_of_insertions = rng.gen_range(number_of_insertions..(1 << (HEIGHT - 1)));
+    let insertions = (number_of_insertions..second_number_of_insertions)
+        .map(|_| rng.gen())
+        .collect::<Vec<_>>();
+    for leaf in insertions {
+        Tree::push(&mut tree, &parameters, &leaf);
+    }
+    let mut cloned_tree = tree.clone();
+    tree.merge_fork_partial(&parameters);
+    cloned_tree.merge_fork(&parameters);
+    assert_eq!(
+        tree.root(),
+        cloned_tree.root(),
+        "Merge fork and merge fork partial should return the same Merkle root"
+    );
```

Comment:
can we have a failing case as well ?

Reply:
also maybe run the test in a loop a few times to get different rngs ?

Reply:
Oh I see what you mean. Yeah I'd say that's out of scope (since this function fails if and only if the old one fails), but some capacity tests for MT might make sense. I'll raise an issue. 

For the loop, yeah I can do 10 times because this test is lightweight.